packages/cli/src/lib/decorators/preconditions/binPreconditionsFactory.ts

```diff
@@ -0,0 +1,127 @@
+import type Command from '@oclif/command';
+import {dedent} from 'ts-dedent';
+import {constants} from 'os';
+import {spawnProcessOutput, SpawnProcessOutput} from '../../utils/process';
+import {satisfies, validRange} from 'semver';
+
+export interface IBinPreconditionsOptions {
+  params?: Array<string>;
+  prettyName?: string;
+  installLink?: string;
+  howToInstallBinText?: string;
+}
+
+const defaultOptions: Required<Omit<IBinPreconditionsOptions, 'prettyName'>> = {
+  params: ['--version'],
+  installLink:
+    'https://github.com/coveo/cli/wiki/Node.js,-NPM-and-NPX-requirements',
+  howToInstallBinText: '',
+};
+
+export function getBinVersionPrecondition(
+  binaryName: string,
+  options?: IBinPreconditionsOptions
+) {
+  const appliedOptions: Required<IBinPreconditionsOptions> = {
+    ...defaultOptions,
+    ...options,
+    prettyName: options?.prettyName || binaryName,
+  };
+  return function (versionRange: string) {
+    return async function (target: Command) {
+      if (!validRange(versionRange)) {
+        target.warn(dedent`
+          Required version invalid: "${versionRange}".
+          Please report this error to Coveo: https://github.com/coveo/cli/issues/new
+        `);
+        return false;
+      }
+
+      const output = await spawnProcessOutput(
+        binaryName,
+        appliedOptions.params
+      );
+
+      if (!isBinInstalled(target, binaryName, appliedOptions, output)) {
+        return false;
+      }
+
+      if (!satisfies(output.stdout, versionRange)) {
```

Comment:
Do you need to strip the `v` prefix before doing the check first?
![image](https://user-images.githubusercontent.com/12199712/114230951-c57d8f00-9947-11eb-938f-c7f71d32061d.png)

Reply:
nope: 
> A leading "=" or "v" character is stripped off and ignored.

https://www.npmjs.com/package/semver